### PR TITLE
Fix local `TraceabilitySuite` configs

### LINKED ISSUE #5464: Local execution of`TraceabilitySuite` fails
# Description
The recent `test-clients` relocation has triggered a couple of problems with running `TraceabilitySuite` locally.

1. In all of the traceability tests, we try to read the initcode of the deployed contract from its corresponding .bin file in order to be able to assert the contract bytecode sidecars. We obtain the path to the file[ using the getResourcePath() method.](https://github.com/hashgraph/hedera-services/blob/e2a60b8f694dd2b874646f6c299318da6e345774/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/Utils.java#L201-L209)  However, on each test we get an exception, e.g.:
```
An error occurred while reading file
java.nio.file.NoSuchFileException: src/main/resource/contract/contracts/Traceability/Traceability.bin
```
  * Why does this happen:
    * [Here, we pass the original ](https://github.com/hashgraph/hedera-services/blob/e2a60b8f694dd2b874646f6c299318da6e345774/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/Utils.java#L204)File[ , constructed from the arguments, to the ](https://github.com/hashgraph/hedera-services/blob/e2a60b8f694dd2b874646f6c299318da6e345774/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/Utils.java#L204)relocatedIfNotPresentInWorkingDir[ method,](https://github.com/hashgraph/hedera-services/blob/e2a60b8f694dd2b874646f6c299318da6e345774/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/Utils.java#L204) which returns the same `File` instance if it's valid, and if not --- a new `File` instance, which is sort of test-clients relocation-aware
    * However, in the return, we do not check whether the returned File is different than the original, and [we always return the original.](https://github.com/hashgraph/hedera-services/blob/e2a60b8f694dd2b874646f6c299318da6e345774/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/Utils.java#L208)

2. In order to verify the sidecar files, we use the SidecarWatcher class in a separate thread, which "watches" the record stream folder and validates each new sidecar file.

   * In the spec properties, we have a property `recordStream.path` --- the path to the record stream folder. [The default value in ](https://github.com/hashgraph/hedera-services/blob/e2a60b8f694dd2b874646f6c299318da6e345774/hedera-node/test-clients/src/main/resource/spec-default.properties#L39)spec-default.properties  specifies the local path, whereas CI [overrides this property in IntegrationTestBase.](https://github.com/hashgraph/hedera-services/blob/e2a60b8f694dd2b874646f6c299318da6e345774/hedera-node/test-clients/src/itest/java/IntegrationTestBase.java#L67)  SidecarWatcher gets passed the[ path to the record stream folder on initialization, based on execution context.](https://github.com/hashgraph/hedera-services/blob/e2a60b8f694dd2b874646f6c299318da6e345774/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/traceability/TraceabilitySuite.java#L5159-L5164)
   * The CI override is correct, that is why CI runs are passing (as they should)
:warning: With the relocation of test-clients, the default local value for the property is incorrect, and the SidecarWatcher does not actually watch the appropriate folder, and it can't verify the sidecars ([the absence of logs from ](https://github.com/hashgraph/hedera-services/blob/e2a60b8f694dd2b874646f6c299318da6e345774/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/verification/traceability/SidecarWatcher.java#L62)SidecarWatcher[ further validate that this is the case](https://github.com/hashgraph/hedera-services/blob/e2a60b8f694dd2b874646f6c299318da6e345774/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/verification/traceability/SidecarWatcher.java#L62))---> local runs will always fail.

# Solution
1. make sure the original path is returned in `getResourcePath()`
2. change `recordStream.path` in spec-default.properties to hedera-node/data/recordstreams/record0.0.3

### PR DESCRIPTION
**Pull request by dimitar-dinev**
**Description**:

See #5464 description

**Related issue(s)**:

Fixes #5464

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/Utils.java [hunk 1]
```diff
@@ -205,7 +205,7 @@ public static String getResourcePath(String resourceName, final String extension
         if (!file.exists()) {
             throw new IllegalArgumentException("Invalid argument: " + path.substring(path.lastIndexOf('/') + 1));
         }
-        return path;
+        return path.equals(file.getPath()) ? path : file.getPath();
```
Comment (Nana-EC) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/Utils.java:208]: Why not just always return `file.getPath()` for simplicity
```suggestion
        return file.getPath();
```

Comment (dimitar-dinev) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/Utils.java]: Indeed 🤦🏽‍♂️  Changed now

File: hedera-node/test-clients/src/main/resource/spec-default.properties [hunk 2]
```diff
@@ -36,7 +36,7 @@ default.node=0.0.3
 default.node.name=DEFAULT_NODE
 default.nodePayment.tinyBars=5000
 default.payer=0.0.2
-recordStream.path=../data/recordstreams/record0.0.3
+recordStream.path=hedera-node/data/recordstreams/record0.0.3
```
Comment (mhess-swl) [inline hedera-node/test-clients/src/main/resource/spec-default.properties:39]: Is this correct? Typically tests in `test-clients` need to be run from the `test-clients` directory, not the `hedera-services` (root) directory

Comment (stoqnkpL) [reply hedera-node/test-clients/src/main/resource/spec-default.properties:39]: It's correct, this was fixed recently - it's no longer required to execute tests from the `test-clients` working directory

**Review by stoqnkpL — APPROVED** [review]
LGTM!

**Review by Nana-EC — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]

**Pull request merged**